### PR TITLE
Fix actionbar trigger key handling

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes https://github.com/microsoft/azuredatastudio/issues/15484

Took this from VS Code actionbar, it basically makes it so that we only trigger on KEY_UP if the actionbar also received the KEY_DOWN as well. This fixes the Notebook scenario since the KEY_DOWN was sent to the callout dialog and so even though the actionbar gets the KEY_UP it won't fire the event. 

Something to remember for the future as well - they've also added functionality to support triggering on KEY_DOWN as well through an option passed in to the constructor. https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/browser/ui/actionbar/actionbar.ts#L90

### CODE REVIEW

**Review by alanrenmsft — APPROVED** [review]
please fix the build error.

File: src/sql/base/browser/ui/taskbar/actionbar.ts [hunk 1]
```diff
@@ -81,7 +84,7 @@ export class ActionBar extends ActionRunner implements IActionRunner {
 			} else if (event.equals(KeyCode.Escape)) {
 				this.cancel();
 			} else if (event.equals(KeyCode.Enter) || event.equals(KeyCode.Space)) {
-				// Nothing, just staying out of the else branch
+				this._triggerKeyDown = true;
```
Comment (chlafreniere) [inline src/sql/base/browser/ui/taskbar/actionbar.ts:87]: There's a problem (not a regression) where pressing escape in the link callout will cause the cell to be unselected as well. Can we fix that at this level, or does that have to be fixed elsewhere?

Comment (Charles-Gagnon) [reply src/sql/base/browser/ui/taskbar/actionbar.ts:87]: That's not a problem we can fix in the actionbar, but likely we could use the same type of fix as this one for that. We can do that separately though - do you have an issue made for that already?

**Comment by Charles-Gagnon** [comment]
Forgot to add a gif showing it fixed, added that now.

**Pull request merged**